week3/homework/src/functions.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+const express = require('express');
+
+const uuid = require('uuid/v4');
+
+const app = express();
+
+const {
+  readFile: _readFile,
+  writeFile: _writeFile
+} = require('fs');
+
+const {
+  promisify
+} = require('util');
+
+const readFile = promisify(_readFile);
+const writeFile = promisify(_writeFile);
+
+
+
+//USE BUILT-IN JSON middleware to automatically parse JSON
+app.use(express.json());
+
+const TODO_FILE = 'todo.json';
+
+function readTodos() {
+  return readFile(TODO_FILE, 'utf-8').then(JSON.parse, () => []);
+}
+
+function writeTodos(todos) {
+  return writeFile(TODO_FILE, JSON.stringify(todos));
+}
+
+//Find Todo
+async function findTodo(req) {
+  const id = req.params.id;
+  const todos = await readTodos();
+  const todo = todos.find(todos => todos.id === id);
+  const done = req.body.done;
+  return await [
+    id,
+    req.body,
+    done,
+    todo,
+    todos
+  ];
```

Comment:
Not sure why you return so many variables 😊 I would have just returned `todo` and `todos`.

Reply:
because i need them in markTodo and updatTodo functions !!!